torch/csrc/TypeInfo.cpp

```diff
@@ -50,16 +50,25 @@ PyObject* THPFInfo_pynew(PyTypeObject* type, PyObject* args, PyObject* kwargs) {
   HANDLE_TH_ERRORS
   static torch::PythonArgParser parser({
       "finfo(ScalarType type)",
+      "finfo()",
   });
+
   torch::ParsedArgs<1> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
-  AT_CHECK(r.idx == 0, "Not a type");
-  at::ScalarType scalar_type = r.scalartype(0);
-  if (!at::isFloatingType(scalar_type)) {
-    return PyErr_Format(
-        PyExc_TypeError,
-        "torch.finfo() requires a floating point input type. Use torch.iinfo to handle '%s'",
-        type->tp_name);
+  AT_CHECK(r.idx < 2, "Not a type");
+  at::ScalarType scalar_type;
+  if (r.idx == 1) {
+    scalar_type = torch::tensors::get_default_tensor_type().scalarType();
+    // The default tensor type can only be set to a floating point type/
+    AT_CHECK(at::isFloatingType(scalar_type));
```

Comment:
nit: AT_ASSERT instead of AT_CHECK.  AT_ASSERT is meant for internal checks.

Reply:
Done